File: Sources/Components/Models/Synthetic APIs/PPVayyarRoom.m
//
//  PPVayyarRoom.m
//  Peoplepower iOS
//
//  Created by Destry Teeter on 7/2/21.
//  Copyright © 2021 People Power Company. All rights reserved.
//

#import "PPVayyarRoom.h"

@implementation PPVayyarRoom

- (id)initWithDeviceId:(NSString *)deviceId
                  xMin: (NSNumber *)xMin
                  xMax: (NSNumber *)xMax
                  yMin: (NSNumber *)yMin
                  yMax: (NSNumber *)yMax
                  zMin: (NSNumber *)zMin
                  zMax: (NSNumber *)zMax {
    self = [super init];
    if (self) {
        self.deviceId = deviceId;
        self.xMin = xMin;
        self.xMax = xMax;
        self.yMin = yMin;
        self.yMax = yMax;
        self.zMin = zMin;
        self.zMax = zMax;
    }
    return self;
}

+ (PPVayyarRoom *)initWithDeviceId:(NSString *)deviceId data:(NSDictionary *)data {
    NSNumber *xMin = @-1.9;
    if ([data objectForKey:@"x_left_meters"]) {
        xMin = [data objectForKey:@"x_left_meters"];
    }
    NSNumber *xMax = @1.9;
    if ([data objectForKey:@"x_right_meters"]) {
        xMax = [data objectForKey:@"x_right_meters"];
    }
    NSNumber *yMin = @0.3;
    if ([data objectForKey:@"y_min_meters"]) {
        yMin = [data objectForKey:@"y_min_meters"];
    }
    NSNumber *yMax = @3.9;
    if ([data objectForKey:@"y_max_meters"]) {
        yMax = [data objectForKey:@"y_max_meters"];
    }
    NSNumber *zMin = @0;
    if ([data objectForKey:@"z_min_meters"]) {
        zMin = [data objectForKey:@"z_min_meters"];
    }
    NSNumber *zMax = @2.5;
    if ([data objectForKey:@"z_max_meters"]) {
        zMax = [data objectForKey:@"z_max_meters"];
    }
    PPVayyarRoom *room = [[PPVayyarRoom alloc] initWithDeviceId:deviceId
                                                           xMin:xMin
                                                           xMax:xMax
                                                           yMin:yMin
                                                           yMax:yMax
                                                           zMin:zMin
                                                           zMax:zMax];
    return room;
}

@end

File: Sources/Components/Models/Synthetic APIs/PPVayyarRoom.swift
//
//  PPVayyarRoom.swift
//  Peoplepower iOS
//
//  Created by Destry Teeter on 7/1/21.
//  Copyright © 2021 People Power Company. All rights reserved.
//

import Foundation

/**
 * # Vayyar Room
 *
 * Parameters
 * - `deviceId` String Device ID to apply these room boundaries to.
 * - `xMin` Float Looking into the room from the device, this is the distance from the center of the device to the left wall of the room in meters. This is a negative number, and if a positive number is given, then it will be turned into a negative number.
 * - `xMax` Float Looking into the room from the device, this is the distance from the center of the device to the right wall of the room in meters. This is a positive number, and if a negative number is given, then it will be turned into a positive number.
 * - `yMin` Float Distance from the Vayyar Home to the opposite wall.
 */
@objc
open class PPVayyarRoom: PPBaseModel {
    
    open override class func primaryKey() -> String? {
        return "deviceId"
    }
    @objc open var deviceId: String!
    
    @objc open var xMin: Double = 1.9
    @objc open var xMax: Double = 1.9
    @objc open var yMin: Double = 0.3
    @objc open var yMax: Double = 3.9
    @objc open var zMin: Double = 0
    @objc open var zMax: Double = 2.5
    
    @objc public convenience init(_ deviceId: String, data: Dictionary<String, Any>) {
        self.init()
        self.deviceId = deviceId
        if let xMin = data["x_min"] as? Double {
            self.xMin = xMin
        }
        if let xMax = data["x_max"] as? Double {
            self.xMax = xMax
        }
        if let yMin = data["y_min"] as? Double {
            self.yMin = yMin
        }
        if let yMax = data["y_max"] as? Double {
            self.yMax = yMax
        }
        if let zMin = data["z_min"] as? Double {
            self.zMin = zMin
        }
        if let zMax = data["z_max"] as? Double {
            self.zMax = zMax
        }
    }
}
